cobigen-templates/templates-oasp4j/src/main/templates/crud_java_server_app/templates/java/${variables.rootPackage}/${variables.component}/dataaccess/api/repo/${variables.entityName}Repository.java.ftl

```diff
@@ -58,9 +59,9 @@ public interface ${variables.entityName}Repository extends DefaultRepository<${v
               }
               
           <#else>
-              ${newFieldType} ${field.name} = criteria.${OaspUtil.resolveIdGetter(field,false,"")};
+              ${newFieldType} ${field.name} = criteria.<#if field.type=='boolean' || field.type=='Boolean'>is${fieldCapName}()<#else>${OaspUtil.resolveIdGetter(field,false,"")}</#if>;
```

Comment:
The search criteria will not contain fields of type boolean as all fields will be boxed to be nullable in case you don't want to set the search criteria parameter.